cyder/base/fields.py

```diff
@@ -51,9 +51,9 @@ def to_python(self, value):
         value = super(MacAddrField, self).to_python(value)
 
         if value:
-            value = value.lower().replace(':', '').replace('-', '')
+            raw = value.lower().replace(':', '').replace('-', '')
             value = reduce(lambda x,y: x + ':' + y,
-                           (value[i:i+2] for i in xrange(0, 12, 2)))
+                           (raw[i:i+2] for i in xrange(0, 12, 2))) + raw[12:]
```

Comment:
Exceptions raised in `to_python` aren't caught by Django, from what I remember, so you just get a server error. I guess the function could return a different string that's obviously an invalid MAC address or something.

But I did completely forget about `str.join`. That should be fixed.